backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/inoutcandidate/api/impl/ReceiptScheduleBL.java

```diff
@@ -752,4 +756,34 @@ public void setBPartnerAddress(final String address)
 			delegate.setBPartnerAddress_Override(address);
 		}
 	}
+
+	@Override
+	public void updateDatePromisedOverrideAndPOReference(@NonNull final PInstanceId pinstanceId, @Nullable final LocalDate datePromisedOverride, @Nullable final String poReference)
+	{
+		if (datePromisedOverride == null && Check.isEmpty(poReference, true))
+		{
+			throw new AdempiereException(MSG_DATEPROMISEDOVERRIDE_POREFERENCE_VALIDATION_ERROR)
+					.markAsUserValidationError();
+		}
+
+		final ICompositeQueryUpdater<I_M_ReceiptSchedule> updater = queryBL
+				.createCompositeQueryUpdater(I_M_ReceiptSchedule.class);
+
+		if (datePromisedOverride != null)
+		{
+			updater.addSetColumnValue(I_M_ReceiptSchedule.COLUMNNAME_DatePromised_Override, datePromisedOverride);
+		}
+
+		if (!Check.isEmpty(poReference, true))
+		{
+			updater.addSetColumnValue(I_M_ReceiptSchedule.COLUMNNAME_POReference, poReference);
+		}
+
+		final int updatedCnt = queryBL.createQueryBuilder(I_M_ReceiptSchedule.class)
+				.setOnlySelection(pinstanceId)
+				.create()
+				.update(updater);
+
+		Loggables.withLogger(logger, Level.INFO).addLog("Updated {} M_ReceiptSchedules", updatedCnt);
```

Comment:
pls return the updateCnt.

not wrong, but usually shall be the job of the caller to do it.
btw, in JavaProcess u already have addLog method.